qiskit/_util.py

```diff
@@ -97,3 +105,57 @@ def _has_connection(hostname, port):
         return True
     except Exception:  # pylint: disable=broad-except
         return False
+
+
+def about():
+    """ Returns information about the Qiskit installation.
+    """
+    hardware_info = local_hardware_info()
+    n_qubits = int(log2(local_hardware_info()['memory']*(1024**3)/16))
+
+    print("")
+    print("Qiskit Details")
+    print("==============")
+    print("Package         Version Number")
+    print("------------------------------")
+    print("Qiskit-terra:       %s" % qiskit.__version__)
+    print("Numpy:              %s" % numpy.__version__)
+    print("Scipy:              %s" % scipy.__version__)
+    print("Networkx:           %s" % nx.__version__)
+    print("Python:             %d.%d.%d" % sys.version_info[0:3])
+    print("")
+    print("Available Providers")
+    print("-------------------")
+    offset = 20
+    for pro in get_available_providers():
+        print(pro[0]+' '*(offset-len(pro[0]))+(pro[1] if pro[1] else ''))
+
+    print("")
+    print("Additional Information")
+    print("----------------------")
+    print("Memory:             %s GB [%s qubits]" % (hardware_info['memory'], n_qubits))
+    print("Number of CPUs:     %s" % hardware_info['cpus'])
+    print("Platform Info:      %s (%s)" % (platform.system(),
+                                           platform.machine()))
+    qiskit_install_path = os.path.dirname(inspect.getsourcefile(qiskit))
+    print("Install path:       %s" % qiskit_install_path)
+
+
+def get_available_providers():
```

Comment:
This would only return providers that are within the `qiskit.providers` namespace, which currently is only BasicAer, Aer, and IBMQ i think.
For example, this 3rd party provider will not be returned?
https://github.com/Qiskit/qiskit-jku-provider

Reply:
This actually is a good match up with #1465 which will add a discovery mechanism for this and populate the providers off of `qiskit.providers`

Reply:
Well, I am not sure how to handle providers that do not inject themselves into qiskit, but rather must be imported.  One could imagine that you could look for all packages with instances of `BaseProvider` in them.  But that seems less than ideal.  There is a PR on provider registration #1465, and this could easily be modified to deal with that case when it comes.